pilot/pkg/bootstrap/server.go

```diff
@@ -1334,3 +1341,14 @@ func (s *Server) serveHTTP() error {
 	s.httpAddr = httpListener.Addr().String()
 	return nil
 }
+
+func serviceUpdateNeedsPush(prev, curr *model.Service) bool {
+	if !features.EnableOptimizedServicePush {
+		return true
+	}
+	if prev == nil {
+		return true
+	}
+	// TODO: Make this more efficient by comparing field by field.
```

Comment:
I am not sure this can be addresssed in a followup? Isn't this super inefficient?

Reply:
Was just wanting to get this approach reviewed and will immediately follow-up with that. I am fine doing it in this same PR if that is better - but are you fine with the approach overall?

Reply:
I guess so. I think there is a better way to do this. I have been thinking on an approach to consistently do this type of operation, all the way from input configs to delta XDS outputs.. hoping to get some time this week to iron it out and maybe present a doc if it works out (likely it will end up not fufilling all my goals :-) ). But that is like super far out, if ever, so no problem doing this.

Reply:
Ok. Cool. I will optimize and update this PR. Wait for your doc :-)